src/ngraph/runtime/gpu/gpu_cuda_kernel_builder.cpp

```diff
@@ -55,6 +55,46 @@ void runtime::gpu::CudaKernelBuilder::get_elementwise_op(codegen::CodeWriter& wr
     return;
 }
 
+void runtime::gpu::CudaKernelBuilder::get_softmax_divide_op(
```

Comment:
I am testing the performance of the elementwise broadcast-divide in softmax with the GPUKernelArgs helper (#1326). This PR implements a better reduce, but I’m wondering where the slowdown on the broadcast divide is occurring and if the new helper improves it. I see a 5x speed up with broadcast, so I expect the broadcast-divide in softmax will also see that improvement, in which case maybe the explicit softmax_divide kernel wouldn’t be necessary, I am checking.

Reply:
Review #1326, realize the main slow down should be passing parameters as array. He if I change to that, I think it will faster.

Reply:
Sorry, I made that patch to your PR branch and with the changes from #1326, and a branch I will push soon to optimize elementwise collective, the performance was comparable.

Reply:
in here, https://gist.github.com/csullivan/92d8bb932e6d68f6caae827e778c2429
what's before and after, are you using Tesla or titan xp?

Reply:
Continuing conversation here instead of the gist comments, this is the MXNet graph I used, it is for batchsize 1024 also, but maybe the graphs are different? Please check with yours, https://gist.githubusercontent.com/csullivan/13988a687a02c657e4567831bd89e44b/raw/6ab47ae04fa0143599dfd4749351ea9051ca649c/mxnet-ngraph-Function_1-MakeForwardFunction-pre-optimized-fprop.json

Reply:
it's different. With your json file.
```
Benchmarking mxnet-ngraph-Function_1-MakeForwardFunction-pre-optimized-fprop.json, GPU backend, 10 iterations.
compile time: 9,246ms
83.9ms per iteration

---- Aggregate times per op type ----
Broadcast     27,466us
Softmax       11,502us
Slice         10,170us
Add            8,745us
Dot            8,615us
Concat         5,440us
Divide         1,786us
Multiply       1,674us
Negative       1,440us
Exp            1,439us
Reshape        1,333us
Result         1,130us
Tanh             939us
```